src/common/functions/src/aggregates/aggregate_combinator_distinct.rs

```diff
@@ -262,22 +136,105 @@ impl AggregateFunction for AggregateDistinctCombinator {
             let netest_place = place.next(layout.size());
             self.nested.drop_state(netest_place);
         }
     }
 }
 
-impl fmt::Display for AggregateDistinctCombinator {
+impl<S, State> fmt::Display for AggregateDistinctCombinator<S, State> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self.nested_name.as_str() {
             "uniq" => write!(f, "uniq"),
             _ => write!(f, "{}_distinct", self.nested_name),
         }
     }
 }
+
+pub fn aggregate_combinator_distinct_desc() -> CombinatorDescription {
+    CombinatorDescription::creator(Box::new(try_create))
+}
+
+pub fn aggregate_combinator_uniq_desc() -> AggregateFunctionDescription {
+    let features = super::aggregate_function_factory::AggregateFunctionFeatures {
+        returns_default_when_only_null: true,
+        ..Default::default()
+    };
+    AggregateFunctionDescription::creator_with_features(Box::new(try_create_uniq), features)
+}
+
+pub fn try_create_uniq(
+    nested_name: &str,
+    params: Vec<DataValue>,
+    arguments: Vec<DataField>,
+) -> Result<Arc<dyn AggregateFunction>> {
+    let creator: AggregateFunctionCreator = Box::new(AggregateCountFunction::try_create);
+    try_create(nested_name, params, arguments, &creator)
+}
+
+pub fn try_create(
+    nested_name: &str,
+    params: Vec<DataValue>,
+    arguments: Vec<DataField>,
+    nested_creator: &AggregateFunctionCreator,
+) -> Result<Arc<dyn AggregateFunction>> {
+    let name = format!("DistinctCombinator({})", nested_name);
+    assert_variadic_arguments(&name, arguments.len(), (1, 32))?;
+
+    let nested_arguments = match nested_name {
+        "count" | "uniq" => vec![],
+        _ => arguments.clone(),
+    };
+    let nested = nested_creator(nested_name, params, nested_arguments)?;
+    if arguments.len() == 1 {
+        let data_type = arguments[0].data_type().clone();
+        let phid = data_type.data_type_id();
+        if phid.is_integer() {
+            with_match_integer_type_id!(phid, |$T| {
+                return Ok(Arc::new(AggregateDistinctCombinator::<
+                    $T,
+                    AggregateDistinctIntegerState<$T>,
+                > {
+                    nested_name: nested_name.to_owned(),
+                    arguments,
+                    nested,
+                    name,
+                    _s: PhantomData,
+                    _state: PhantomData,
+                }));
+            }, {
+                panic!("GetKVActionReply");
+            })
+        }
+        if phid.is_string() {
+            return Ok(Arc::new(AggregateDistinctCombinator::<
+                Vec<u8>,
+                AggregateDistinctStringState,
+            > {
+                nested_name: nested_name.to_owned(),
+                arguments,
+                nested,
+                name,
+                _s: PhantomData,
+                _state: PhantomData,
+            }));
+        }
+        if phid.is_floating() {}
```

Comment:
did not cover float.

Reply:
😅 Sorry, I missed some commits when I was using cherry pick, I'll commit them later with the other fixes